management/management.go

```diff
@@ -127,18 +127,19 @@ func New(domain, clientID, clientSecret string, options ...apiOption) (*Manageme
 		option(m)
 	}
 
-	oauth2 := client.OAuth2(m.url, clientID, clientSecret)
+	if m.http == nil {
+		oauth2 := client.OAuth2(m.url, clientID, clientSecret)
 
-	_, err = oauth2.Token(m.ctx)
-	if err != nil {
-		return nil, err
+		_, err = oauth2.Token(m.ctx)
+		if err != nil {
+			return nil, err
+		}
+		m.http = client.New(m.ctx, oauth2)
```

Comment:
I am not sure whether we should completely forego authentication entirely if a custom client has been used. Perhaps we can create new or wrap depending on the options supplied?

If possible, something like this would be nice imo:

```go
if m.http == nil {
	m.http = client.New(m.ctx, m.oauth2)
} else {
	m.http = client.Wrap(m.http, m.ctx, m.oauth2)
}
```

Reply:
>  Perhaps we can create new or wrap depending on the options supplied?

Good idea. I'll come up with something for review.

> If possible, something like this would be nice imo:
> 
> ```go
> if m.http == nil {
> 	m.http = client.New(m.ctx, m.oauth2)
> } else {
> 	m.http = client.Wrap(m.http, m.ctx, m.oauth2)
> }
> ```
Great, thanks for the example. 🙇